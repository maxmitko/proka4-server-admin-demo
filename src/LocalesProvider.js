import { locales } from 'locales'
import React from 'react';

export const LocalesContext = React.createContext();

export default props => <LocalesContext.Provider value={locales} children={props.children} />