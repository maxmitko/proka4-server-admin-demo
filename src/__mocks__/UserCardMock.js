export default {
    userCard: {
        data: [
            {
                id: 2,
                count: 11,
                price: 1500,
                debt: 0,
                isActive: 1,
                purchaseTime: '2019-01-28T16:36:53.000Z',
                user: {
                    id: 2,
                    username: 'Natalia',
                    fullname: 'Филатова Флорентина Леонидовна',
                    email: 'kras@mail.ru',
                    phone: '',
                    money: 670,
                    isActive: 1,
                    regDate: '2017-10-12T18:00:00.000Z',
                    lastAct: '2019-04-08T19:01:22.000Z'
                },
                service: {
                    id: 2,
                    title: 'Бодифлекс',
                    description: 'description 1',
                    isActive: 1,
                    myOrder: 1,
                    image: '0hpTp3jsx.jpeg'
                },
                visits: [
                    {
                        id: 445,
                        visitStatus: 1,
                        regDate: '2019-02-05T21:00:00.000Z',
                        regTime: '08:20:00'
                    },
                    {
                        id: 486,
                        visitStatus: 1,
                        regDate: '2019-02-20T21:00:00.000Z',
                        regTime: '08:20:00'
                    },
                    {
                        id: 489,
                        visitStatus: 1,
                        regDate: '2019-03-11T21:00:00.000Z',
                        regTime: '08:30:00'
                    },
                    {
                        id: 491,
                        visitStatus: 0,
                        regDate: '2019-03-30T21:00:00.000Z',
                        regTime: '18:00:00'
                    },
                    {
                        id: 495,
                        visitStatus: 1,
                        regDate: '2019-04-01T21:00:00.000Z',
                        regTime: '09:00:00'
                    },
                    {
                        id: 504,
                        visitStatus: 1,
                        regDate: '2019-04-01T21:00:00.000Z',
                        regTime: '08:15:00'
                    },
                    {
                        id: 506,
                        visitStatus: 1,
                        regDate: '2019-03-31T21:00:00.000Z',
                        regTime: '08:15:00'
                    }
                ]
            },
            {
                id: 66,
                count: 8,
                price: 1600,
                debt: 0,
                isActive: 1,
                purchaseTime: '2019-01-28T12:23:49.000Z',
                user: {
                    id: 2,
                    username: 'Natalia',
                    fullname: 'Филатова Флорентина Леонидовна',
                    email: 'kras@mail.ru',
                    phone: '',
                    money: 670,
                    isActive: 1,
                    regDate: '2017-10-12T18:00:00.000Z',
                    lastAct: '2019-04-08T19:01:22.000Z'
                },
                service: {
                    id: 5,
                    title: 'Тренинг "познай тело"',
                    description: 'description 2',
                    isActive: 1,
                    myOrder: 6,
                    image: 'dhA1V76uj.jpeg'
                },
                visits: [
                    {
                        id: 464,
                        visitStatus: 1,
                        regDate: '2019-02-11T21:00:00.000Z',
                        regTime: '08:20:00'
                    },
                    {
                        id: 476,
                        visitStatus: 1,
                        regDate: '2019-02-17T21:00:00.000Z',
                        regTime: '08:20:00'
                    }
                ]
            },
            {
                id: 92,
                count: 9,
                price: 420,
                debt: 0,
                isActive: 1,
                purchaseTime: '2019-02-20T03:52:30.000Z',
                user: {
                    id: 2,
                    username: 'Natalia',
                    fullname: 'Филатова Флорентина Леонидовна',
                    email: 'kras@mail.ru',
                    phone: '',
                    money: 670,
                    isActive: 1,
                    regDate: '2017-10-12T18:00:00.000Z',
                    lastAct: '2019-04-08T19:01:22.000Z'
                },
                service: {
                    id: 4,
                    title: 'Королевская осанка',
                    description: 'description 3',
                    isActive: 1,
                    myOrder: 4,
                    image: 'oZto6R1M5.jpeg'
                },
                visits: [
                    {
                        id: 500,
                        visitStatus: 1,
                        regDate: '2019-04-01T21:00:00.000Z',
                        regTime: '17:50:00'
                    }
                ]
            },
            {
                id: 93,
                count: 6,
                price: 700,
                debt: 0,
                isActive: 1,
                purchaseTime: '2019-02-20T03:52:32.000Z',
                user: {
                    id: 2,
                    username: 'Natalia',
                    fullname: 'Филатова Флорентина Леонидовна',
                    email: 'kras@mail.ru',
                    phone: '',
                    money: 670,
                    isActive: 1,
                    regDate: '2017-10-12T18:00:00.000Z',
                    lastAct: '2019-04-08T19:01:22.000Z'
                },
                service: {
                    id: 4,
                    title: 'Королевская осанка',
                    description: 'description 4',
                    isActive: 1,
                    myOrder: 4,
                    image: 'oZto6R1M5.jpeg'
                },
                visits: [
                    {
                        id: 501,
                        visitStatus: 1,
                        regDate: '2019-04-01T21:00:00.000Z',
                        regTime: '17:50:00'
                    }
                ]
            },
            {
                id: 94,
                count: 9,
                price: 800,
                debt: 0,
                isActive: 1,
                purchaseTime: '2019-02-20T03:52:35.000Z',
                user: {
                    id: 2,
                    username: 'Natalia',
                    fullname: 'Филатова Флорентина Леонидовна',
                    email: 'kras@mail.ru',
                    phone: '',
                    money: 670,
                    isActive: 1,
                    regDate: '2017-10-12T18:00:00.000Z',
                    lastAct: '2019-04-08T19:01:22.000Z'
                },
                service: {
                    id: 4,
                    title: 'Королевская осанка',
                    description: 'description 5',
                    isActive: 1,
                    myOrder: 4,
                    image: 'oZto6R1M5.jpeg'
                },
                visits: [
                    {
                        id: 502,
                        visitStatus: 1,
                        regDate: '2019-04-01T21:00:00.000Z',
                        regTime: '17:50:00'
                    }
                ]
            }
        ],
        totalCount: 0,
        options: {},
        error: false,
        loading: false,
        crud: {
            data: [],
            error: false,
            loading: false
        }
    }
}