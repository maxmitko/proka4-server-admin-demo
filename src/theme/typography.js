
const defaultFontFamily = "'Roboto', sans-serif"

export default {
    h1: {
        fontWeight: 300,
        fontSize: 6,
        letterSpacing: -1.5,
        lineHeight: 1,
        fontFamily: defaultFontFamily,
        tag: "h2",
    },
    h2: {
        fontWeight: 300,
        fontSize: 3.75,
        letterSpacing: -0.5,
        lineHeight: 1,
        fontFamily: defaultFontFamily,
        tag: "h2",
    },
    h3: {
        fontWeight: 400,
        fontSize: 3,
        letterSpacing: 0,
        lineHeight: 1.04,
        fontFamily: defaultFontFamily,
        tag: "h3",
    },
    h4: {
        fontWeight: 400,
        fontSize: 2.125,
        letterSpacing: 0.25,
        lineHeight: 1.17,
        fontFamily: defaultFontFamily,
        tag: "h4",
    },
    h5: {
        fontWeight: 400,
        fontSize: 1.5,
        letterSpacing: 0,
        lineHeight: 1.33,
        fontFamily: defaultFontFamily,
        tag: "h5",
    },
    h6: {
        fontWeight: 300,
        fontSize: 1.15,
        letterSpacing: 0,
        lineHeight: 1.4,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    subtitle1: {
        fontWeight: 400,
        fontSize: 1,
        letterSpacing: 0.15,
        lineHeight: 1.75,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    subtitle2: {
        fontWeight: 500,
        fontSize: 0.875,
        letterSpacing: 0.1,
        lineHeight: 1.57,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    body1: {
        fontWeight: 400,
        fontSize: 1,
        letterSpacing: 0.5,
        lineHeight: 1.5,
        fontFamily: defaultFontFamily,
        tag: "p",
    },
    body2: {
        fontWeight: 400,
        fontSize: 0.875,
        letterSpacing: 0.25,
        lineHeight: 1.5,
        fontFamily: defaultFontFamily,
        tag: "p",
    },
    button: {
        fontWeight: 400,
        fontSize: 0.875,
        letterSpacing: 1.25,
        lineHeight: 1.5,
        fontFamily: "'Roboto Condensed', sans-serif",
        tag: "span",
    },
    caption: {
        fontWeight: 400,
        fontSize: 0.75,
        letterSpacing: 0.4,
        lineHeight: 1.66,
        fontFamily: defaultFontFamily,
        tag: "span",
    },
    overline: {
        fontWeight: 400,
        fontSize: 0.75,
        letterSpacing: 1.5,
        lineHeight: 2.66,
        fontFamily: defaultFontFamily,
        tag: "span",
    },
    inherit: {
        fontWeight: "inherit",
        fontSize: "inherit",
        letterSpacing: "inherit",
        lineHeight: "inherit",
        fontFamily: "inherit",
        tag: "span",
    },
}