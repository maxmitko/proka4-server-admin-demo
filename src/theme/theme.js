import typography from './typography'
import produce from "immer"

export const theme = {
    palette: {
        primary: "hsla(0, 0%, 0%, 1)",
        secondary: "hsla(187, 100%, 42%, 1)",
        background: "hsla(0, 0%, 93%, 1)",
        surface: "hsla(0, 0%, 100%, 1)",
        error: "hsla(0, 100%, 42%, 1)",
    },
    paletteOn: {
        primary: "hsla(0, 0%, 100%, 1)",
        secondary: "hsla(0, 0%, 100%, 1)",
        background: "hsla(0, 0%, 0%, 1)",
        surface: "hsla(210, 9%, 31%, 1)",
        error: "hsla(0, 0%, 100%, 1)",
    },
    extra: {
        default: null,
        gray: "hsla(0, 0%, 76%, 1)",
        primary: "hsla(291, 64%, 42%, 1)",
        danger: "hsla(4, 90%, 58%, 1)",
        success: "hsla(123, 36%, 52%, 1)",
        warning: "hsla(34, 98%, 51%, 1)",
        info: "hsla(187, 100%, 42%, 1)",
        rose: "hsla(340, 82%, 58%, 1)",
        white: "hsla(0, 0%, 100%, 1)",
    },
    extraOn: {
        default: null,
        gray: "hsla(0, 0%, 100%, 1)",
        primary: "hsla(0, 0%, 100%, 1)",
        danger: "hsla(0, 0%, 100%, 1)",
        success: "hsla(0, 0%, 100%, 1)",
        warning: "hsla(0, 0%, 100%, 1)",
        info: "hsla(0, 0%, 100%, 1)",
        rose: "hsla(0, 0%, 100%, 1)",
        white: "hsla(0, 0%, 0%, 1)",
    },
    extraLight: {
        default: null,
        gray: "hsla(0, 0%, 72%, 1)",
        primary: "hsla(291, 88%, 57%, 1)",
        danger: "hsla(22, 100%, 53%, 1)",
        success: "hsla(123, 45%, 57%, 1)",
        warning: "hsla(34, 98%, 60%, 1)",
        info: "hsla(187, 95%, 46%, 1)",
        rose: "hsla(340, 82%, 65%, 1)",
        white: "hsl(0, 100%, 100%)",
    },
    extraShadow: {
        default: null,
        gray: "hsla(0, 0%, 30%, 0.22)",
        primary: "hsla(291, 64%, 54%, 0.25)",
        danger: "hsla(0, 100%, 58%, 0.25)",
        success: "hsla(123, 45%, 57%, 0.25)",
        warning: "hsla(34, 98%, 60%, 0.25)",
        info: "hsla(187, 95%, 46%, 0.25)",
        rose: "hsla(340, 82%, 63%, 0.25)",
        white: "hsla(0, 0%, 30%, 0.25)",
    },
    typography: typography,
    zIndex: {
        mobileStepper: 1000,
        appBar: 1100,
        drawer: 1200,
        modal: 1300,
        snackbar: 1400,
        tooltip: 1500,
    }
}

const newTheme = produce(theme, draftTheme => {
    draftTheme.extra.default = theme.extra.primary
    draftTheme.extraOn.default = theme.extraOn.primary
    draftTheme.extraLight.default = theme.extraLight.primary
    draftTheme.extraShadow.default = theme.extraShadow.primary
})


export default newTheme