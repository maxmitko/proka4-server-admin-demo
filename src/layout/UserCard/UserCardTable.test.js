import LocalesProvider from 'LocalesProvider'
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import { createStore } from 'redux'
import { StoreContext } from 'redux-react-hook';
import { ThemeProvider } from 'styled-components'
import theme from 'theme/theme'

import UserCardMock from '../../__mocks__/UserCardMock'
import TestComponent from './UserCardTable'

beforeAll(() => {
    const modalRoot = global.document.createElement('div');
    modalRoot.setAttribute('id', 'modal-root');
    const body = global.document.querySelector('body');
    body.appendChild(modalRoot);
});

const mockStore = createStore((state, action) => UserCardMock);

const withHOC = ui => {
    return (
        <StoreContext.Provider value={mockStore}>
            <LocalesProvider>
                <ThemeProvider theme={theme}>
                    {ui}
                </ThemeProvider >
            </LocalesProvider>
        </StoreContext.Provider>
    )
}

const props = {
    userId: 5
}

describe('renderer UserCardTable', () => {
    let wrapper;
    let element;
    
    beforeEach(() => {
        element = withHOC(<TestComponent {...props} />)
    })

    afterEach(() => {
        wrapper && wrapper.unmount();
    });

    it('should render correctly', () => {
        const tree = renderer.create(element).toJSON();
        expect(tree).toMatchSnapshot();
    });


    it('row count should render correctly', () => {
        wrapper = mount(element);
        expect(wrapper.find('tr')).toHaveLength(6);
    });
});
