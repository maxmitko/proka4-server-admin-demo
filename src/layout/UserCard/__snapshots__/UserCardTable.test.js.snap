// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer UserCardTable should render correctly 1`] = `
Array [
  <div
    className="sc-hzDkRC fZwiDW"
  >
    <div
      className="sc-eHgmQL jvQohW"
    >
      <div
        className="sc-jWBwVP gdAsra"
      />
      <table
        className="sc-cvbbAY bisXyk"
      >
        <thead>
          <tr>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              №
            </td>
            <td>
              Курс
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Стоимость
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Долг
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Квота
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Остаток
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Дата покупки
            </td>
            <td>
              
            </td>
          </tr>
        </thead>
        <tbody>
          <tr
            className="sc-hSdWYo cVheuo"
            onClick={[Function]}
          >
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              1
            </td>
            <td>
              Бодифлекс
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              1500
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              0
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              11
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              5
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              28-01-2019
            </td>
            <td>
              <div
                className="sc-iAyFgw eaCSUU"
              >
                <button
                  className="sc-bdVaJa kpKQwF"
                  color="success"
                  onClick={[Function]}
                  size="medium"
                >
                  <div
                    className="sc-htpNat hostrS"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-1"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </div>
                  <div
                    className="rtr-root sc-bwzfXH dsGjGK"
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    timeout={
                      Object {
                        "enter": 500,
                        "exit": 500,
                      }
                    }
                  >
                    <span />
                  </div>
                </button>
                <button
                  className="sc-bdVaJa fsVEsd"
                  color="danger"
                  onClick={[Function]}
                  size="medium"
                >
                  <div
                    className="sc-htpNat hostrS"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-1"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </div>
                  <div
                    className="rtr-root sc-bwzfXH dsGjGK"
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    timeout={
                      Object {
                        "enter": 500,
                        "exit": 500,
                      }
                    }
                  >
                    <span />
                  </div>
                </button>
              </div>
            </td>
          </tr>
          <tr
            className="sc-hSdWYo cVheuo"
            onClick={[Function]}
          >
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              2
            </td>
            <td>
              Тренинг "познай тело"
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              1600
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              0
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              8
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              6
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              28-01-2019
            </td>
            <td>
              <div
                className="sc-iAyFgw eaCSUU"
              >
                <button
                  className="sc-bdVaJa kpKQwF"
                  color="success"
                  onClick={[Function]}
                  size="medium"
                >
                  <div
                    className="sc-htpNat hostrS"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-1"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </div>
                  <div
                    className="rtr-root sc-bwzfXH dsGjGK"
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    timeout={
                      Object {
                        "enter": 500,
                        "exit": 500,
                      }
                    }
                  >
                    <span />
                  </div>
                </button>
                <button
                  className="sc-bdVaJa fsVEsd"
                  color="danger"
                  onClick={[Function]}
                  size="medium"
                >
                  <div
                    className="sc-htpNat hostrS"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-1"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </div>
                  <div
                    className="rtr-root sc-bwzfXH dsGjGK"
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    timeout={
                      Object {
                        "enter": 500,
                        "exit": 500,
                      }
                    }
                  >
                    <span />
                  </div>
                </button>
              </div>
            </td>
          </tr>
          <tr
            className="sc-hSdWYo cVheuo"
            onClick={[Function]}
          >
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              3
            </td>
            <td>
              Королевская осанка
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              420
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              0
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              9
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              8
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              20-02-2019
            </td>
            <td>
              <div
                className="sc-iAyFgw eaCSUU"
              >
                <button
                  className="sc-bdVaJa kpKQwF"
                  color="success"
                  onClick={[Function]}
                  size="medium"
                >
                  <div
                    className="sc-htpNat hostrS"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-1"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </div>
                  <div
                    className="rtr-root sc-bwzfXH dsGjGK"
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    timeout={
                      Object {
                        "enter": 500,
                        "exit": 500,
                      }
                    }
                  >
                    <span />
                  </div>
                </button>
                <button
                  className="sc-bdVaJa fsVEsd"
                  color="danger"
                  onClick={[Function]}
                  size="medium"
                >
                  <div
                    className="sc-htpNat hostrS"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-1"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </div>
                  <div
                    className="rtr-root sc-bwzfXH dsGjGK"
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    timeout={
                      Object {
                        "enter": 500,
                        "exit": 500,
                      }
                    }
                  >
                    <span />
                  </div>
                </button>
              </div>
            </td>
          </tr>
          <tr
            className="sc-hSdWYo cVheuo"
            onClick={[Function]}
          >
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              4
            </td>
            <td>
              Королевская осанка
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              700
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              0
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              6
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              5
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              20-02-2019
            </td>
            <td>
              <div
                className="sc-iAyFgw eaCSUU"
              >
                <button
                  className="sc-bdVaJa kpKQwF"
                  color="success"
                  onClick={[Function]}
                  size="medium"
                >
                  <div
                    className="sc-htpNat hostrS"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-1"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </div>
                  <div
                    className="rtr-root sc-bwzfXH dsGjGK"
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    timeout={
                      Object {
                        "enter": 500,
                        "exit": 500,
                      }
                    }
                  >
                    <span />
                  </div>
                </button>
                <button
                  className="sc-bdVaJa fsVEsd"
                  color="danger"
                  onClick={[Function]}
                  size="medium"
                >
                  <div
                    className="sc-htpNat hostrS"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-1"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </div>
                  <div
                    className="rtr-root sc-bwzfXH dsGjGK"
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    timeout={
                      Object {
                        "enter": 500,
                        "exit": 500,
                      }
                    }
                  >
                    <span />
                  </div>
                </button>
              </div>
            </td>
          </tr>
          <tr
            className="sc-hSdWYo cVheuo"
            onClick={[Function]}
          >
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              5
            </td>
            <td>
              Королевская осанка
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              800
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              0
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              9
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              8
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              20-02-2019
            </td>
            <td>
              <div
                className="sc-iAyFgw eaCSUU"
              >
                <button
                  className="sc-bdVaJa kpKQwF"
                  color="success"
                  onClick={[Function]}
                  size="medium"
                >
                  <div
                    className="sc-htpNat hostrS"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-1"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </div>
                  <div
                    className="rtr-root sc-bwzfXH dsGjGK"
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    timeout={
                      Object {
                        "enter": 500,
                        "exit": 500,
                      }
                    }
                  >
                    <span />
                  </div>
                </button>
                <button
                  className="sc-bdVaJa fsVEsd"
                  color="danger"
                  onClick={[Function]}
                  size="medium"
                >
                  <div
                    className="sc-htpNat hostrS"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-1"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </div>
                  <div
                    className="rtr-root sc-bwzfXH dsGjGK"
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    timeout={
                      Object {
                        "enter": 500,
                        "exit": 500,
                      }
                    }
                  >
                    <span />
                  </div>
                </button>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  <div
    className="sc-bRBYWo hodFoK"
  >
    <div
      className="sc-fjdhpX dZTHV"
      color="default"
      data={Array []}
      label={null}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Выбирите абонимент"
      value=""
    >
      <label
        className="sc-jTzLTM dDUuVs"
      />
      <div
        className="sc-gqjmRU elixZU"
        color="default"
        onClick={[Function]}
      >
        <div
          className="sc-cSHVUG fkomLr"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-1"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20.57 14.86L22 13.43 20.57 12 17 15.57 8.43 7 12 3.43 10.57 2 9.14 3.43 7.71 2 5.57 4.14 4.14 2.71 2.71 4.14l1.43 1.43L2 7.71l1.43 1.43L2 10.57 3.43 12 7 8.43 15.57 17 12 20.57 13.43 22l1.43-1.43L16.29 22l2.14-2.14 1.43 1.43 1.43-1.43-1.43-1.43L22 16.29z"
            />
          </svg>
        </div>
        <input
          className="sc-jzJRlG geYCmJ"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Выбирите абонимент"
          value=""
        />
        <div
          className="sc-kAzzGY dVttjy"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-1"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </div>
        <div
          className="sc-VigVT kdlbmq"
          color="default"
        />
      </div>
    </div>
    <button
      className="sc-bdVaJa bRuARb"
      color="default"
      disabled={true}
      onClick={[Function]}
      size="medium"
      type="submit"
    >
      <div
        className="sc-htpNat hostrS"
      >
        Добавить
      </div>
      <div
        className="rtr-root sc-bwzfXH dsGjGK"
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        timeout={
          Object {
            "enter": 500,
            "exit": 500,
          }
        }
      >
        <span />
      </div>
    </button>
  </div>,
]
`;
